### ISSUE TITLE
Toolbar menu doesn't work with out-of-process WebExtensions enabled

### ISSUE THREAD

**Issue by emmyemi** [mozilla/testpilot-containers#670]
The `extensions.webextensions.remote` pref has been flipped to `true` by default in Nightly for Windows, but this breaks the toolbar menu, making it impossible to interact with. This happens regardless of whether the icon is placed in the toolbar, the overflow menu, or the tab bar. Screenshot below:

![image](https://user-images.githubusercontent.com/14862440/28196234-47c53bee-681d-11e7-8d70-767f7e221ddb.png)

**Comment by jonathanKingston** [comment]
Hey could you provide details of your setup please. If you check
about:support page and paste in any extensions you have and what OS you are
on etc.
Thanks

**Comment by jonathanKingston** [comment]
Looks like this is the Windows default now for Nightly.

I am seeing the following when debugging this in windows.
```
JPM [error] JavaScript error: state is undefined, line 90: Error: Failed to retrieve the identities or variation. We cannot continue.
```

**Comment by jonathanKingston** [comment]
@groovecoder still looking into this, it seems like `extensions.webextensions.remote` breaks any legacy addon message passing that use promises.

I have managed to repair the `queryIdentitiesState` message however there are others, the next to be seen is: `getTabs`

Unfortunately the only fix I have found for this is the following:
- popup -> background sendMessage
- background -> index sendMessage
- index -> background (this is through onConnect listener in background)
- background -> popup (again through an onConnect)
- popup has to respond to the existing code with a Promise which setTimeouts the response from a object prop that the message has set

I am trying to see if I can reliably cut out steps in this and maybe I can write a wrapper that does it for all the async methods so its easier to check it works for all. This is at least going to take till the end of the day I expect though.

**Comment by jonathanKingston** [comment]
Managed to get it to:
- popup -> index sendMessage
- index -> background (this is through onConnect listener in background)
- background -> popup (again through an onConnect)
- popup has to respond to the existing code with a Promise which setTimeouts the response from a object prop that the message has set

**Comment by jonathanKingston** [comment]
Ths also seems to be impacting async messages to the background script.

@andymckay do you know if this remote pref can be flipped off for Windows?

**Comment by jonathanKingston** [comment]
@groovecoder and @bakulf I have finished all methods for both index.js and background.js that has a promise. Both pass the message back.

**Comment by jonathanKingston** [comment]
Looks like it is just 56 Windows at the moment that is getting this: http://searchfox.org/mozilla-central/source/browser/app/profile/firefox.js#88

We should make sure @groovecoder they are not in the Shield study. //cc @johngruen

**Comment by andymckay** [comment]
Its Windows on Nightly only and the pref is flippable to off.

**Comment by jonathanKingston** [comment]
This looks like it should be fixed in Windows Nightly soon based on this bug here: https://bugzilla.mozilla.org/show_bug.cgi?id=1381023

**Comment by emmyemi** [comment]
I can confirm that this has been fixed as of Nightly build 2017-07-15.

**Comment by moi12518** [comment]
I've been searching through the issues list for something similar to my problem and this is the closest I've gotten by far. For quite some time, I've had one issue with Containers. The icon on the toolbar works and allows me to set up Containers for only a few sessions after Containers are enabled before it stops working entirely. This isn't drastically impacting my usage, as after I recognized the pattern of 'enable Containers, a few sessions pass, and then the icon is useless,' I disabled and re-enabled Containers to create all the Containers I could see myself using and haven't needed to add any since. However, I can imagine that someone else could also have this problem and it be a serious one, so I'm speaking my bit now (as I finally found out I could report problems). 

![image](https://user-images.githubusercontent.com/30208004/28245498-2146f6aa-69bd-11e7-887d-b869d674139d.png)

Not sure if that worked, so I'll describe the picture I tried to put in just in case. The icon looks normal when I don't to anything to it, but when I click it, only a white dropdown arrow appears. There is no popup/bubble thing at all. This is replicable and happens every time I try to click the icon in every session. I have AdBlock Plus, uBlock Origin, and Ghostery, so it's possible one of them or a combination thereof is the source of the problem. (There's also Blur which just appeared one day in a Firefox update, so if it's the problem then I'll be surprised.) I've considered that having a Firefox Persona may be important, but it was a brief consideration.

I know this isn't the same problem, but it's similar enough that they might be a bit related. This issue has existed for months and had I realized I could have reported it sooner, I would have.
